packages/core/src/external/carequality/ihe-gateway-v2/outbound/schema.ts

```diff
@@ -0,0 +1,18 @@
+import { z } from "zod";
+
+export const schemaOrEmpty = <T extends z.ZodTypeAny>(schema: T) =>
+  z.union([schema, z.literal("")]);
+export const schemaOrArray = <T extends z.ZodTypeAny>(schema: T) =>
+  z.union([schema, z.array(schema)]);
+export const schemaOrArrayOrEmpty = <T extends z.ZodTypeAny>(schema: T) =>
+  z.union([schema, z.array(schema), z.literal("")]);
```

Comment:
the whole concept of the z.literal("") is that sometimes we get xmls with tags like this

<telecom /> which the xml parser translates to 
{ telecom: "" }

so we want to be able to handle those cases and not fail the schema validation if we get that